File: src/api/java/CharSort.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    CharSort.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/ 

package com.microsoft.z3;

/**
 * A Character sort
 **/
public class CharSort extends Sort
{
    CharSort(Context ctx, long obj)
    {
        super(ctx, obj);
    }

    CharSort(Context ctx) { super(ctx, Native.mkCharSort(ctx.nCtx())); {  }}

}

File: src/api/dotnet/CharSort.cs
/*++
Copyright (c) 2016 Microsoft Corporation

Module Name:

    CharSort.cs

Abstract:

    Z3 Managed API: Character Sorts

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/

using System.Diagnostics;
using System;

namespace Microsoft.Z3
{
    /// <summary>
    ///  A Character sort
    /// </summary>
    public class CharSort : Sort
    {
        #region Internal
        internal CharSort(Context ctx, IntPtr obj) : base(ctx, obj) { Debug.Assert(ctx != null); }
        internal CharSort(Context ctx) : base(ctx, Native.Z3_mk_char_sort(ctx.nCtx)) { Debug.Assert(ctx != null); }
        #endregion
    }
}
